evm_loader/program/src/types/transaction.rs

```diff
@@ -117,6 +135,299 @@ impl rlp::Decodable for Transaction {
+    gas_limit: U256,
+    target: Option<Address>,
+    value: U256,
+    call_data: crate::evm::Buffer,
+    r: U256,
+    s: U256,
+    chain_id: U256,
+    recovery_id: u8,
+    access_list: Vec<(Address, Vec<StorageKey>)>,
+    rlp_len: usize,
+    hash: [u8; 32],
+    signed_hash: [u8; 32],
+}
+
+impl rlp::Decodable for AccessListTx {
+    fn decode(rlp: &rlp::Rlp) -> Result<Self, rlp::DecoderError> {
+        let rlp_len = {
+            let info = rlp.payload_info()?;
+            info.header_len + info.value_len
+        };
+
+        if rlp.as_raw().len() != rlp_len {
+            return Err(rlp::DecoderError::RlpInconsistentLengthAndData);
+        }
+
+        let chain_id: U256 = u256(&rlp.at(0)?)?;
+        let nonce: u64 = rlp.val_at(1)?;
+        let gas_price: U256 = u256(&rlp.at(2)?)?;
+        let gas_limit: U256 = u256(&rlp.at(3)?)?;
+        let target: Option<Address> = {
+            let target = rlp.at(4)?;
+            if target.is_empty() {
+                if target.is_data() {
+                    None
+                } else {
+                    return Err(rlp::DecoderError::RlpExpectedToBeData);
+                }
+            } else {
+                Some(target.as_val()?)
+            }
+        };
+
+        let value: U256 = u256(&rlp.at(5)?)?;
+        let call_data = crate::evm::Buffer::from_slice(rlp.at(6)?.data()?);
+
+        let rlp_access_list = rlp.at(7)?;
+        let mut access_list = vec![];
+
+        for entry in rlp_access_list.iter() {
+            // Check if entry is a list
+            if entry.is_list() {
+                // Parse address from first element
+                let address: Address = entry.at(0)?.as_val()?;
+
+                // Get storage keys from second element
+                let mut storage_keys: Vec<StorageKey> = vec![];
+
+                for key in entry.at(1)?.iter() {
+                    storage_keys.push(key.as_val()?);
+                }
+
+                access_list.push((address, storage_keys));
+            }
+        }
+
+        let y_parity: u8 = rlp.at(8)?.as_val()?;
+        let r: U256 = u256(&rlp.at(9)?)?;
+        let s: U256 = u256(&rlp.at(10)?)?;
+
+        if rlp.at(11).is_ok() {
+            return Err(rlp::DecoderError::RlpIncorrectListLen);
+        }
+
+        let hash = solana_program::keccak::hashv(&[&[0x01], rlp.as_raw()]).to_bytes();
+        let signed_hash = eip2718_signed_hash(&[0x01], rlp, 8)?;
+
+        let tx = AccessListTx {
+            nonce,
+            gas_price,
+            gas_limit,
+            target,
+            value,
+            call_data,
+            r,
+            s,
+            chain_id,
+            recovery_id: y_parity,
+            access_list,
+            rlp_len,
+            hash,
+            signed_hash,
+        };
+
+        Ok(tx)
+    }
+}
+
+// TODO: Will be added as a part of EIP-1559
+// struct DynamicFeeTx {}
+
+// TODO: Will be added as a part of EIP-1559
+// struct BlobTx {}
```

Comment:
Shard Blob transactions are described in EIP-4844. 
And they would probably never be added.

Reply:
removed any references to blobs